addons/survey/models/survey_survey.py

```diff
@@ -378,6 +380,12 @@ def _onchange_survey_type(self):
                 'scoring_type': 'scoring_with_answers',
             })
 
+    @api.constrains('scoring_type', 'users_can_go_back')
+    def _check_scoring_after_page_availability(self):
+        for survey in self:
+            if survey.scoring_type == 'scoring_with_answers_after_page' and survey.users_can_go_back:
+                raise ValidationError(_("You cannot use roaming when scoring type is scoring with answers after each page."))
```

Comment:
Continuing Florian's comment: something like
```suggestion
        failing = self.filtered(
            lambda survey: survey.scoring_type == 'scoring_with_answers_after_page' and survey.users_can_go_back
        )
        if failing:
            raise ValidationError(
                _("Combining roaming and 'scoring with answers after each page' is not possible; please update following surveys: %(survey_names)s",
                  survey_names=", ".join(failing.mapped('name'))
            )
```